### PR TITLE
Error ranking 2

### PR DESCRIPTION
**Pull request by kirti-d**
the PR contains all the js involved in functioning of result page

### CODE REVIEW

File: src/main/webapp/onClickEvents.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// if on 2nd slide of carousel 
+addclass = () => {
```
Comment (shudhatmajain) [inline src/main/webapp/onClickEvents.js:2]: elaborate function name

File: src/main/webapp/createDOM.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    // if fileType is error then we need to add 
+    // btn to show stackTrace 
+    if(fileType == ERRORS){
+      addStackTraceBtn(liElement, logError);
+    }
+    return liElement;
+}
+
+// add button for stack trace
+addStackTraceBtn = (liElement, logError) => {
```
Comment (shudhatmajain) [inline src/main/webapp/createDOM.js:32]: let's have a separate file for stack traces

File: src/main/webapp/createDOM.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+// display file not found message on UI
+fileNotFound = () => {
```
Comment (shudhatmajain) [inline src/main/webapp/createDOM.js:2]: let's have a separate file related to carousel functionalities.

File: src/main/webapp/pagination.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    const start = recordsPerPage * ((page - 1) % noOfPages);
+    let end = start + (recordsPerPage - 1);
+    if(page == lastPage) {
+        end = start + (noOfRecordsOnLastPage - 1);
+    } 
+    return {START: start, END: end};
+}
+
+// return page no to be fetched from database
+getPageToBeFetched = () => {
```
Comment (shudhatmajain) [inline src/main/webapp/pagination.js:108]: Let's have a separate file for maintaining page data.

File: src/main/webapp/result.css [hunk 5]
```diff
@@ -0,0 +1,130 @@
+/* Body Header */
+#body-header {
```
Comment (shudhatmajain) [inline src/main/webapp/result.css:2]: Let's put this in separate PR.

File: src/main/webapp/pagination.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+//  calculating starting index and size of block to be fetched for log file 
+getPageStartAndSizeForLogFile = (page) => {
+    const start = ((page - 1) * recordsPerPage);
+    let size = recordsPerPage;
+    if( page == 1) {
+        size = (noOfPages * recordsPerPage);
+    } 
+    return {START: start, SIZE: size};
```
[the remark this answers is not in the document]
Comment (kirti-d) [reply src/main/webapp/pagination.js:71]: start and size was being used at other places too so just moved them to constant file
